test/integration/singlecluster/controller/core/workload_controller_test.go

```diff
@@ -544,3 +546,97 @@ var _ = ginkgo.Describe("Workload controller", ginkgo.Ordered, ginkgo.ContinueOn
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Workload controller with resource retention", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	var (
+		ns              *corev1.Namespace
+		createdWorkload kueue.Workload
+		localQueue      *kueue.LocalQueue
+		clusterQueue    *kueue.ClusterQueue
+		flavor          *kueue.ResourceFlavor
+	)
+
+	ginkgo.BeforeAll(func() {
+		fwk.StartManager(
+			ctx, cfg,
+			managerAndControllerSetup(
+				&config.Configuration{
+					ObjectRetentionPolicies: &config.ObjectRetentionPolicies{
+						Workloads: &config.WorkloadRetentionPolicy{
+							AfterFinished: &metav1.Duration{
+								Duration: util.ShortTimeout,
+							},
+						},
+					},
+				},
+			),
+		)
+
+		gomega.Expect(features.SetEnable(features.ObjectRetentionPolicies, true)).To(gomega.Succeed())
+	})
+
+	ginkgo.AfterAll(func() {
+		fwk.StopManager(ctx)
+	})
+
+	ginkgo.BeforeEach(func() {
+		ns = util.CreateNamespaceFromPrefixWithLog(ctx, k8sClient, "core-workload-")
+	})
+
+	ginkgo.AfterEach(func() {
+		createdWorkload = kueue.Workload{}
+		clusterQueue = nil
+		localQueue = nil
+		flavor = nil
```

Comment:
This should not be needed. If the variables are scoped to When, let's place them there

Reply:
Moved to the respective When cases